tests/test_detrend.py

```diff
@@ -89,6 +89,16 @@ def test_detrend(show=False):
     x += 2 * np.sin(2 * np.pi * np.arange(1000) / 200)[:, None]
     y, _, _ = detrend(x, 5, basis='sinusoids', show=True)
 
+    # trial-masked detrending
+    trend = np.linspace(0, 100, 1000)[:, None]
+    data = 3 * np.random.randn(*trend.shape)
+    data[:100, :] = 100
+    x = trend + data
+    events = np.arange(30, 970, 40)
```

Comment:
I haven't looked into it in detail, but since your events are entirely equidistant from one another, the "masked weight" is not going to have much of an effect.  

In other words, the trend inferred from the masked data is pretty much exactly the same. 

If we want a more striking difference between the trial mask and the vanilla implementation we would probably need a slightly more sophisticated simulated dataset.

Reply:
For instance, maybe we could add some pseudo-ERPs during the events, and see how it might mess up the detrending if we do not use a trial-mask

Reply:
Consider the slightly more complex example, with slightly longer 'events' to be masked out :

```python 

# trial-masked detrending
trend = np.cumsum(np.random.randn(1000, 1) + 0.1, axis=0)
data = 3 * np.random.randn(*trend.shape)
x = trend + data
events = np.arange(30, 970, 200)
tmin, tmax, sfreq = -1., 2., 20  # 3s 'events' 
for e in events:
    x[e - 10:e + 30] += 20  # add some significant deviations from baseline during events

# without trial-mask
w = np.ones(x.shape)
y1, _, _ = detrend(x, 10, w, basis='polynomials', show=show)

# with trial-mask
w = create_masked_weight(x, events, tmin, tmax, sfreq)
y2, _, _ = detrend(x, 10, w, basis='polynomials', show=show)

f, ax = plt.subplots(2, sharex=True,
                        gridspec_kw={'height_ratios': [.9, .1]})
ax[0].plot(y1, label='no trial-mask')
ax[0].plot(y2, label='trial-mask')
ax[0].legend()
ax[1].pcolormesh(w.T)
plt.show()

``` 

This produces very different detrending results : 

<img width="761" alt="Screenshot 2021-05-12 at 15 25 07" src="https://user-images.githubusercontent.com/10333715/117982380-3d1f4f00-b336-11eb-9c0c-f0d0edf346d6.png">

I'm not sure which is necessarily better, but the two lines are different, meaning the trial-mask is clearly doing **something**

Reply:
Sorry for the delay. It is hard to find time working on it right now. Will try in the next coming weeks. 
I will try on a real MEG dataset to see if it's worth or not. What do you think is the next step?

Reply:
To me this is fine, so I would merge. We can always revisit this issue in a future PR.

Reply:
Sounds good ! Thank you